lib/rules/jsx-pascal-case.js

```diff
@@ -6,19 +6,60 @@
 'use strict';
 
 const elementType = require('jsx-ast-utils/elementType');
-const XRegExp = require('xregexp');
 const docsUrl = require('../util/docsUrl');
 const jsxUtil = require('../util/jsx');
 
-// ------------------------------------------------------------------------------
-// Constants
-// ------------------------------------------------------------------------------
+function testDigit(char) {
+  const charCode = char.charCodeAt(0);
+  return charCode >= 48 && charCode <= 57;
+}
+
+function testUpperCase(char) {
+  const upperCase = char.toUpperCase();
+  return char === upperCase && upperCase !== char.toLowerCase();
+}
 
-// eslint-disable-next-line no-new
-const hasU = (function hasU() { try { new RegExp('o', 'u'); return true; } catch (e) { return false; } }());
+function testLowerCase(char) {
+  const lowerCase = char.toLowerCase();
+  return char === lowerCase && lowerCase !== char.toUpperCase();
+}
 
-const PASCAL_CASE_REGEX = XRegExp('^(.*[.])*([\\p{Lu}]|[\\p{Lu}]+[\\p{Ll}0-9]+(?:[\\p{Lu}0-9]+[\\p{Ll}0-9]*)*)$', hasU ? 'u' : '');
-const ALL_CAPS_TAG_REGEX = XRegExp('^[\\p{Lu}0-9]+([\\p{Lu}0-9_]*[\\p{Lu}0-9]+)?$', hasU ? 'u' : '');
+function testPascalCase(name) {
+  if (!testUpperCase(name.charAt(0))) {
+    return false;
+  }
+  const {length} = name;
+  let atLeastOneLowerCase = false;
+  for (let i = 1; i < length; i += 1) {
```

Comment:
caching the length in a for loop in JS is slower, don't try to outsmart the engine.
```suggestion
  let atLeastOneLowerCase = false;
  for (let i = 1; i < name.length; i += 1) {
```

Reply:
should perhaps this loop be a `some` instead of a loop?

Reply:
If speed is a concern, I don't know that `some` will help.
However it's not something I care a lot about; I would find `some` more readable.

I'll write it with `some` and you'll tell me if I should keep it or revert.

Reply:
I just attempted it and:
- We don't start at `0`. The logic in the loop would make it work even if we were because there is the right overlap between what we test on the first character and what we test on the rest, but that would be confusing to read.
- String doesn't have the `some` method. I would need to do a `.split()` first, which would probably come at a cost.

I'll still do it if you want me to, though.

Reply:
Nah, fair point, a loop is fine.